### PR TITLE
Scrub compaction task

### PR DESCRIPTION
**Pull request by Deexie**
Task manager's tasks covering scrub compaction on top, 
shard and table level. 

For this levels we have common scrub tasks for each scrub
mode since they share code. Scrub modes will be differentiated
on compaction group level.

### CODE REVIEW

**Comment by Deexie** [comment]
@bhalevy review please

File: api/storage_service.cc [hunk 1]
```diff
@@ -1500,10 +1500,10 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
                     auto& cm = db.get_compaction_manager();
                     auto& cf = db.find_column_family(keyspace, cfname);
                     sstables::compaction_stats stats{};
-                    co_await cf.parallel_foreach_table_state([&] (compaction::table_state& ts) mutable -> future<> {
+                    co_await cf.parallel_foreach_table_state(coroutine::lambda([&] (compaction::table_state& ts) mutable -> future<> {
```
Comment (bhalevy) [inline api/storage_service.cc]: Why is it needed?

Comment (Deexie) [reply api/storage_service.cc]: To keep reference to `stats` alive

File: compaction/task_manager_module.cc [hunk 2]
```diff
@@ -128,4 +128,44 @@ future<> shard_upgrade_sstables_compaction_task_impl::run() {
...
+}
+
+tasks::is_internal table_scrub_sstables_compaction_task_impl::is_internal() const noexcept {
+    return tasks::is_internal::yes;
+}
+
+future<> table_scrub_sstables_compaction_task_impl::run() {
+    auto& cm = _db.get_compaction_manager();
+    auto& cf = _db.find_column_family(_status.keyspace, _status.table);
+    co_await cf.parallel_foreach_table_state(coroutine::lambda([&] (compaction::table_state& ts) mutable -> future<> {
```
Comment (bhalevy) [inline compaction/task_manager_module.cc:164]: here too, I don't see why a coroutine::lambda is needed.
it has no capture by values and it's being use in a coroutine frame anyway

**Comment by Deexie** [comment]
- drop seastar::coroutine
- rebase

**Review by denesb — APPROVED** [review]

**Pull request merged**